### ISSUE TITLE
Answer Image Subfolder Issue

### ISSUE THREAD

**Issue by Konsider** [Jefferson49/joomla4_plugin_sexy_polling#106]
When choosing an answer's image, if the image is held in a subfolder on the site, it doesn't work. It is possible to browse to the subfolder and select the image, but then the image is simply not displayed. Images only seem to work if in the root image folder. Of course, this is not very practical.

**Comment by Jefferson49** [comment]
There is a bug in the code, if the image file does not end with a simple file extension. If I choose a file in the answers editor, I get something like: "**images/banners/LogoLeft.gif#joomlaImage://local-images/banners/LogoLeft.gif?width=146&height=76**", where after the file extension (.gif) follows some other text. In my test environment, this results in a PHP error. However, the image is still shown. It might be different in other PHP configurations. Maybe, the image is not shown on your server in this case.

If possible, you can try to change the following line of code:
https://github.com/Jefferson49/Joomla_extension_sexypolling_reloaded/blob/joomla_5.x/com_sexypolling/site/helpers/helper.php#L409

Replace the following code: 
`$img_path_cache = $matches[1] . '-tmb-w' . $poll_data->img_width . '.' . $matches[2];`
with this more robust code: 
`$img_path_cache = ($matches[1] ?? '') . '-tmb-w' . $poll_data->img_width . '.' . ($matches[2] ?? '');`

On your Joomla installation on the server, you can find this file in the following path:
**joomla\components\com_sexypolling\helpers\helper.php**

**Comment by Konsider** [comment]
This would need to be changed by you, so that in future updates, changes by the public are not overwritten by the 'official' releases.

The issue happens for me with the simple .jpg extension. It does seem a bit strange that it adds that additional info into the file path. The file path should just be a straight path to the image. If it did that, it would probably work fine for all supported extensions.

**Comment by Jefferson49** [comment]
> This would need to be changed by you, so that in future updates, changes by the public are not overwritten by the 'official' releases.

Yes, I will change it and release a new version with the bugfix.

The background why I asked you to check in advance is that I am not sure if the identified change will solve your issue. The identied bug explains why there is a PHP error. However, it does not necessarily explain why the picture is not shown.

>The issue happens for me with the simple .jpg extension

Does that mean, you do not have any further text after ".jpg"? Can you post a screenshot how it looks in the sexy polling answer editor? 

For me, it looks like this. The text after "#" causes the problem:

![answer_image](https://github.com/user-attachments/assets/7f674c4c-40dd-4c66-8389-d964ec6593fc)

**Comment by Konsider** [comment]
Ah, I think I've found the issue. At least on my side here. I had my images already in a sub folder of the plugin called Event Gallery. So, when I browse to them in the picker and choose one, it simply doesn't show. I assumed it is because it's in a sub folder.

However, if I create a new folder in the images root, then a sub folder within that folder and use an image within that, it works fine! I also tried picking an image from another set of folders, and it's fine!

So, it seems for some crazy reason, I just can't use the Event Gallery existing set of folders. So, it shouldn't be an issue for anyone else really, so it's not worth bothering about any further.

Sorry about that, I should have made sure it wasn't some crazy thing like that on my end before I opened the issue :)
